lib/common.ts

```diff
@@ -0,0 +1,286 @@
+import assert from 'assert';
+import path from 'path';
+
+export const STORE_BLOB = 0;
+export const STORE_CONTENT = 1;
+export const STORE_LINKS = 2;
+export const STORE_STAT = 3;
+export const ALIAS_AS_RELATIVE = 0; // require("./file.js") // file or directory
+export const ALIAS_AS_RESOLVABLE = 1; // require("package")
+
+const win32 = process.platform === 'win32';
+const hasURL = typeof URL !== 'undefined';
+
+function uppercaseDriveLetter(f: string) {
+  if (f.slice(1, 3) !== ':\\') return f;
+  return f[0].toUpperCase() + f.slice(1);
+}
+
+function removeTrailingSlashes(f: string) {
+  if (f === '/') {
+    return f; // dont remove from "/"
+  }
+
+  if (f.slice(1) === ':\\') {
+    return f; // dont remove from "D:\"
+  }
+
+  let last = f.length - 1;
+
+  while (true) {
+    const char = f.charAt(last);
+
+    if (char === '\\') {
+      f = f.slice(0, -1);
+      last -= 1;
+    } else if (char === '/') {
+      f = f.slice(0, -1);
+      last -= 1;
+    } else {
+      break;
+    }
+  }
+  return f;
+}
+
+export function isRootPath(p: any) {
```

Comment:
Looks like there's a few type `any` checks missing. Add them here or in a follow up? Up to you.

![image](https://user-images.githubusercontent.com/9113740/112570964-0bb5e880-8db5-11eb-9e89-bbf48d90757c.png)